lib/DDG/Spice/OpenNIC.pm

```diff
@@ -0,0 +1,81 @@
+package DDG::Spice::OpenNIC;
+# ABSTRACT: Displays The nearest OpenNIC DNS servers
+
+use DDG::Spice;
+spice is_cached => 1;
+
+# Metadata
+name "OpenNIC";
+source "OpenNIC API";
+description "Search for OpenNIC DNS using some criteras";
+primary_example_queries "opennic", "neutral dns" , "censor-free dns";
+secondary_example_queries "opennic 42.42.42.42", "opennic ipv6", "opennic ipv6 42.42.42.42";
+category "cheat_sheets";
+topics "sysadmin";
+
+# Code info
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/OpenNIC.pm";
+attribution github  => ["cylgom", "cylgom"], # it's me
+            web     => ['http://cylgom.net', 'cylgom'],
+            twitter => "cylgom";
+            
+# OpenNIC API endpoint
+# We prepare a 'spice from' with the place for 4 arguments
+spice from => '([^/]+)/?(?:([^/]+)/?(?:([^/]+)/?(?:([^/]+)/?(?:([^/]+)|)|)|)|)';
+spice to => 'https://api.opennicproject.org/geoip/?json&lat=$1&lon=$2&ipv=$3&res=$4&ip=$5';
+spice wrap_jsonp_callback => 1;
+
+# Triggers
+triggers any => "neutral dns", "opennic", "open nic", "censor free dns";
+
+# Handle statement
+handle remainder => sub {
+    my $ipv; 
+    my $ip;
+    my $lat;
+    my $lon;
+    my $number;
+
+    # ip version
+    # user asked ipv4 (or ip4)
+    if ($_ =~ /ipv?4/) {
+    $ipv='4';
+    }
+    # user asked ipv6 (or ip6)
+    elsif ($_ =~ /ipv?6/) {
+    $ipv='6';
+    }
+    # user didn't ask => collect both
+    else {
+    $ipv="all";
+    }
+
+    # target ip for detection :
+    # user-forced (ipv4 regex from stackoverflow, id : 53497)
+    if ($_ =~ /(((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/){ 
```

Comment:
There's IPV6 in one of the example queries above. Should this handle IPV6 addresses too?